doc/article/en-US/router-configuration.md

```diff
@@ -403,12 +403,12 @@ A pipeline step must be an object that contains a `run(navigationInstruction, ne
     }
   </source-code>
   <source-code lang="TypeScript">
-    import {Redirect, NavigationInstruction, RouterConfiguration, Next} from 'aurelia-router';
+    import {RouterConfiguration, NavigationInstruction, Next, Redirect} from 'aurelia-router';
 
     export class App {
       configureRouter(config: RouterConfiguration): void {
-        config.title = 'Aurelia';
-        config.addAuthorizeStep(AuthorizeStep);
+        const step = new AuthorizeStep;
```

Comment:
What happened here?

Reply:
First, the title configuration is removed because it is not needed for this example.
Second, the `addAuthorizeStep` function expects as argument an object with a `run` method. The `AuthorizeStep` is a class so it does not satisfy that requirement bacause it contains a step method under the prototype (`AuthorizeStep.prototype.run`). Since, an instance of the class is created which is what `addAuthorizeStep` expects.

Reply:
I see. This does work, but I think in the docs we should prefer relying on the Aurelia DI. That is, we should pass the AuthorizeStep class as a key and let Aurelia DI instantiate it. In an application it isn't significant, but everyone is going to copy paste this code so its best we stick to best practices and conventions.